pkg/acsengine/defaults_test.go

```diff
@@ -497,6 +497,108 @@ func TestStorageProfile(t *testing.T) {
 
 }
 
+// TestMasterProfileDefaults covers tests for setMasterProfileDefaults
+func TestMasterProfileDefaults(t *testing.T) {
+	mockCS := getMockBaseContainerService("1.10.3")
+	properties := mockCS.Properties
+	properties.OrchestratorProfile.OrchestratorType = "Kubernetes"
+	properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet = ""
+	properties.OrchestratorProfile.KubernetesConfig.NetworkPlugin = "azure"
+	properties.MasterProfile.AvailabilityProfile = ""
+	properties.MasterProfile.Count = 3
+	mockCS.Properties = properties
+	setPropertiesDefaults(&mockCS, false, false)
+	if properties.MasterProfile.IsVirtualMachineScaleSets() {
+		t.Fatalf("Master VMAS, AzureCNI: MasterProfile AvailabilityProfile did not have the expected default configuration, got %s, expected %s",
+			properties.MasterProfile.AvailabilityProfile, api.AvailabilitySet)
+	}
+	if properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet != DefaultKubernetesSubnet {
+		t.Fatalf("Master VMAS, AzureCNI: MasterProfile ClusterSubnet did not have the expected default configuration, got %s, expected %s",
+			properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet, DefaultKubernetesSubnet)
+	}
+	if properties.MasterProfile.Subnet != properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet {
+		t.Fatalf("Master VMAS, AzureCNI: MasterProfile Subnet did not have the expected default configuration, got %s, expected %s",
+			properties.MasterProfile.Subnet, properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet)
+	}
+	if properties.AgentPoolProfiles[0].Subnet != properties.MasterProfile.Subnet {
+		t.Fatalf("Master VMAS, AzureCNI: AgentPoolProfiles Subnet did not have the expected default configuration, got %s, expected %s",
+			properties.AgentPoolProfiles[0].Subnet, properties.MasterProfile.Subnet)
+	}
+	if properties.MasterProfile.FirstConsecutiveStaticIP != "10.255.255.5" {
+		t.Fatalf("Master VMAS, AzureCNI: MasterProfile FirstConsecutiveStaticIP did not have the expected default configuration, got %s, expected %s",
+			properties.MasterProfile.FirstConsecutiveStaticIP, "10.255.255.5")
+	}
+
+	properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet = ""
```

Comment:
it looks like we're starting a new test here, so let's reset our mock object here (`properties = mockCS.Properties`); or, just make it a new test that is explicit that we're testing masterProfile +VMSS